File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03281/p03281_num6_parsed.py
import math

def count_odd_numbers_with_eight_divisors(N):
    count = 0
    for i in range(1, N+1):
        if is_odd_number_with_eight_divisors(i):
            count += 1
    return count

def is_odd_number_with_eight_divisors(num):
    divisors = 0
    for i in range(1, int(math.sqrt(num))+1):
        if num % i == 0:
            divisors += 2
        if divisors > 8:
            return False
    if math.sqrt(num) == int(math.sqrt(num)):
        divisors -= 1
    if num % 2 != 0 and divisors == 8:
        return True
    return False

N = int(input())

print(count_odd_numbers_with_eight_divisors(N))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03281/p03281_num6_parsed.cpp
#include <iostream>

int countOddNumbersWithEightDivisors(int N) {
    int count = 0;
    for (int i = 1; i <= N; i++) {
        int divisors = 0;
        for (int j = 1; j <= i; j++) {
            if (i % j == 0) {
                divisors++;
            }
        }
        if (divisors == 8 && i % 2 != 0) {
            count++;
        }
    }
    return count;
}

int main() {
    int N;
    std::cin >> N;

    int result = countOddNumbersWithEightDivisors(N);
    std::cout << result;

    return 0;
}